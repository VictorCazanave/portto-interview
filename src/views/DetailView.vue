<template>
	<div>
		Detail View

		<template v-if="asset">
			<span>{{ asset.collectionName }}</span>
			<AssetPreview :asset="asset" />
			<span>
				{{ asset.description }}
			</span>

			<a :href="asset.permalink">Permalink</a>
		</template>

		<div v-else>
			ERROR
		</div>

	</div>
</template>

<script>
import { mapState } from 'vuex'
import AssetPreview from '@/components/AssetPreview.vue'

// TODO: Handle fallback (request or localStorage) when reload
export default {
	name: 'DetailView',
	components: {
		AssetPreview,
	},
	computed: {
		...mapState(['assets']),

		asset() {
			return this.assets.find((asset) => asset.tokenId === this.$route.params.tokenId)
		},
	},
}
</script>

<style lang="scss">
</style>
