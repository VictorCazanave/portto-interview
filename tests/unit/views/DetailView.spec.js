import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import DetailView from '@/views/DetailView.vue'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('DetailView', () => {
	const assets = [
		{
			tokenId: 'tokenId0',
			name: 'name0',
			imageUrl: 'imageUrl0',
			description: 'description0',
			permalink: 'permalink0',
			collectionName: 'collectionName0',
		},
		{
			tokenId: 'tokenId1',
			name: 'name1',
			imageUrl: 'imageUrl1',
			description: 'description1',
			permalink: 'permalink1',
			collectionName: 'collectionName1',
		},
	]
	const store = new Vuex.Store({
		state: { assets },
	})
	const wrapper = shallowMount(DetailView, {
		store,
		localVue,
		mocks: {
			$route: {
				params: {
					tokenId: 'tokenId1',
				},
			},
		},
		stubs: ['router-link'],
	})

	it('renders asset data properly', () => {
		expect(wrapper.element).toMatchSnapshot()
	})
})
