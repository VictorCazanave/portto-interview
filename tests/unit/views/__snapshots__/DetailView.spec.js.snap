// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailView renders asset data properly 1`] = `
<div
  class="page"
>
  <header
    class="page-header header"
  >
    <router-link-stub
      class="previous"
      title="Go to assets list"
      to="[object Object]"
    >
      
			&lt;
		
    </router-link-stub>
     
    <h1>
      
			collectionName1
		
    </h1>
  </header>
   
  <main
    class="page-body"
  >
    <div
      class="page-content"
    >
      <img
        alt="name1"
        class="image"
        src="imageUrl1"
      />
       
      <span
        class="name"
      >
        
				name1
			
      </span>
       
      <p
        class="description"
      >
        
				description1
			
      </p>
    </div>
  </main>
   
  <a
    class="permalink"
    href="permalink1"
    rel="”noopener”"
    target="_blank"
  >
    
		Permalink
	
  </a>
</div>
`;
